src/rgw/driver/sfs/object.cc

```diff
@@ -502,19 +502,20 @@ void SFSObject::_refresh_meta_from_object() {
     if (db_version.has_value()) {
       auto uuid = objref->path.get_uuid();
       auto deleted = db_version->object_state == ObjectState::DELETED;
-      objref = std::make_shared<sfs::Object>(get_name(), uuid, deleted);
-      objref->version_id = db_version->id;
+      objref.reset(std::move(sfs::Object::create_for_query(get_name(), uuid, deleted,
+                                                           db_version->id)));
```

Comment:
All the `create_for_*` functions return a raw pointer. 
Why calling `std::move` on a raw pointer? I think it adds confusion to the reader. 
It would make sense if `create_for_*` functions were retuning a `std::unique_ptr`...

Reply:
good catch. will remove